packages/react-components/src/components/Modal/ModalHeader.tsx

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Icon, IconProps } from '../Icon';
+import styles from './Modal.module.scss';
+import { cx } from '@emotion/css';
+
+const baseClass = 'modal-header';
+
+interface ModalHeaderProps {
+  title?: React.ReactNode;
+  iconProps?: IconProps;
+  children?: React.ReactNode;
+  className?: 'string';
+}
+
+export const ModalHeader: React.FC<ModalHeaderProps> = ({
+  title,
+  iconProps,
+  children,
+  className = '',
+}) => {
+  const mergedClassNames = cx(styles[baseClass], className);
+
+  return (
+    <div className={mergedClassNames}>
+      {iconProps && (
+        <Icon className={styles[`${baseClass}__heading-icon`]} {...iconProps} />
+      )}
+      <div className={styles[`${baseClass}__heading-header`]}>
```

Comment:
Why is the header component wrapping the description? Or should it be named differently (I'm writing about class name)?